### PR TITLE
internal click-repl commands (exit / general help for repl usage)

### PR DESCRIPTION
**Pull request by cosminbasca**
add six as an explicit dependency for Py2 / 3 compatibility (already part of prompt_toolkit)

### CODE REVIEW

**Comment by cosminbasca** [comment]
will introduce internal commands available after ":"

**Comment by cosminbasca** [comment]
the "internal command" functionality allows one to type ":q" to exit the repl, a nice feature is to help a ":h" option where the general usage of the repl is echoed

**Comment by untitaker** [comment]
Please file those as separate pull requests, for real this time.

It's very hard to review those changes when they're interwoven like that.

**Comment by untitaker** [comment]
Yikes, sorry about that comment. I get why those changes are related now.

**Comment by cosminbasca** [comment]
I'm not sure how to separate those changes as they all build on top of the previous commit, sadly I cannot investigate this now as I have already spent too much time on it.

File: click_repl/__init__.py [hunk 1]
```diff
@@ -7,6 +9,54 @@
 import os
 import shlex
 import sys
+import six
+from .exceptions import InternalCommandException, ExitReplException
+
+__internal_commands__ = dict()
```
Comment (untitaker) [inline click_repl/__init__.py]: Please do `_internal_commands`.

File: click_repl/__init__.py [hunk 2]
```diff
@@ -7,6 +9,54 @@
...
+
+
+def _register_internal_command(names, target, description=None):
+    assert hasattr(target, '__call__'), 'internal command must be a callable'
+    global __internal_commands__
```
Comment (untitaker) [inline click_repl/__init__.py]: Not necessary.

**Comment by untitaker** [comment]
@cosminbasca Yeah, sorry. This one is fine. Do you have time to make the changes mentioned in the diff?

File: click_repl/__init__.py [hunk 3]
```diff
@@ -7,6 +9,54 @@
...
+        return target_info[0]
+    return default
+
+
+def _exit_internal():
+    raise ExitReplException()
+
+
+def _help_internal():
+    global __internal_commands__
```
Comment (untitaker) [inline click_repl/__init__.py]: Not necessary.

File: click_repl/__init__.py (hunk 1 above)
Comment (cosminbasca) [inline click_repl/__init__.py]: makes sense, done

File: click_repl/__init__.py (hunk 2 above)
Comment (cosminbasca) [inline click_repl/__init__.py]: are you referring to the assert?

Comment (untitaker) [inline click_repl/__init__.py]: No, the `global` statement.

You can modify the dictionary just fine without `global`. You do need it when you want to bind a different object to that name.

Comment (cosminbasca) [inline click_repl/__init__.py]: ah :) yes you are right, an automatic explicit from my side. Will remove it

File: click_repl/__init__.py [hunk 4]
```diff
@@ -7,6 +8,53 @@
...
+        return target_info[0]
+    return default
+
+
+def _exit_internal():
+    raise ExitReplException()
+
+
+def _help_internal():
+    global _internal_commands
```
Comment (untitaker) [inline click_repl/__init__.py]: a second one that is not necessary.

**Comment by untitaker** [comment]
Excellent, thanks!

**Comment by cosminbasca** [comment]
:+1:

**Pull request merged**